libr/anal/cc.c

```diff
@@ -211,6 +211,30 @@ R_API void r_anal_cc_set_error(RAnal *anal, const char *convention, const char *
 	r_strbuf_fini (&sb);
 }
 
+#if R2_580
+R_API int r_anal_cc_max_arg(RAnal *anal, const char *cc, void *oldDB, char *oldCC, int *oldArg) {
+	int i = 0;
+	r_return_val_if_fail (anal && DB && cc, 0);
+	*oldDB = NULL;
+	*oldCC = NULL;
+	*oldArg = 0;
+	if (oldDB == DB && !strcmp (cc, oldCC)) {
+		return oldArg;
+	}
+	oldDB = DB;
```

Comment:
anal->ccmax_cache.db = DB;
free (anal->ccmax_cache.cc );
anal->ccmax_cache.cc = strdup(cc);

Reply:
in r_anal_free () { you need to free the anal->ccmax_cache.cc)